.buildkite/scripts/cloud-cleanup.sh

```diff
@@ -0,0 +1,120 @@
+      -e ACCOUNT_SECRET="${ELASTIC_PACKAGE_AWS_SECRET_KEY}" \
+      -e ACCOUNT_KEY="${ELASTIC_PACKAGE_AWS_ACCESS_KEY}" \
+      -e ACCOUNT_PROJECT="${ELASTIC_PACKAGE_AWS_USER_SECRET}" \
+      -e CREATION_DATE="${DELETE_RESOURCES_BEFORE_DATE}" \
+      "${CLOUD_REAPER_IMAGE}" \
+        cloud-reaper \
+          --config /etc/cloud-reaper/config.yml \
+          validate
+
+    echo "Scanning resources"
+    docker run --rm -v $(pwd)/.buildkite/configs/cleanup.aws.yml:/etc/cloud-reaper/config.yml \
+      -e ACCOUNT_SECRET="${ELASTIC_PACKAGE_AWS_SECRET_KEY}" \
+      -e ACCOUNT_KEY="${ELASTIC_PACKAGE_AWS_ACCESS_KEY}" \
+      -e ACCOUNT_PROJECT="${ELASTIC_PACKAGE_AWS_USER_SECRET}" \
+      -e CREATION_DATE="${DELETE_RESOURCES_BEFORE_DATE}" \
+      "${CLOUD_REAPER_IMAGE}" \
+        cloud-reaper \
+          --config /etc/cloud-reaper/config.yml \
+          ${COMMAND} | tee "${AWS_RESOURCES_FILE}"
+}
+
+cloud_reaper_gcp() {
+    echo "Validating configuration"
+    docker run --rm -v $(pwd)/.buildkite/configs/cleanup.gcp.yml:/etc/cloud-reaper/config.yml \
+      -e ACCOUNT_SECRET="${ELASTIC_PACKAGE_GCP_KEY_SECRET}" \
+      -e ACCOUNT_KEY="${ELASTIC_PACKAGE_GCP_EMAIL_SECRET}" \
+      -e ACCOUNT_PROJECT="${ELASTIC_PACKAGE_GCP_PROJECT_SECRET}" \
+      -e CREATION_DATE="${DELETE_RESOURCES_BEFORE_DATE}" \
+      "${CLOUD_REAPER_IMAGE}" \
+        cloud-reaper \
+          --config /etc/cloud-reaper/config.yml \
+          validate
+
+    echo "Scanning resources"
+    docker run --rm -v $(pwd)/.buildkite/configs/cleanup.gcp.yml:/etc/cloud-reaper/config.yml \
+      -e ACCOUNT_SECRET="${ELASTIC_PACKAGE_GCP_KEY_SECRET}" \
+      -e ACCOUNT_KEY="${ELASTIC_PACKAGE_GCP_EMAIL_SECRET}" \
+      -e ACCOUNT_PROJECT="${ELASTIC_PACKAGE_GCP_PROJECT_SECRET}" \
+      -e CREATION_DATE="${DELETE_RESOURCES_BEFORE_DATE}" \
+      "${CLOUD_REAPER_IMAGE}" \
+        cloud-reaper \
+          --config /etc/cloud-reaper/config.yml \
+          ${COMMAND} | tee "${GCP_RESOURCES_FILE}"
+}
+
+echo "--- Cleaning up GCP resources older than ${DELETE_RESOURCES_BEFORE_DATE}..."
+cloud_reaper_gcp
+
+if any_resources_to_delete "${GCP_RESOURCES_FILE}"; then
+    resources_to_delete=1
+fi
+
+echo "--- Cleaning up AWS resources older than ${DELETE_RESOURCES_BEFORE_DATE}..."
+cloud_reaper_aws
+
+if ! any_resources_to_delete "${AWS_RESOURCES_FILE}" ; then
+    resource_to_delete=1
+fi
+
+if [ "${resource_to_delete}" -eq 1 ]; then
+    message="There are resources to be deleted"
+    echo "${message}"
+    if running_on_buildkite ; then
+         buildkite-agent annotate \
+             "${message}" \
+             --context "ctx-cloud-reaper-error" \
+             --style "error"
+    fi
+    exit 1
+fi
+
+# TODO: List and delete the required resources using aws cli
+echo "--- Cleaning up other AWS resources older than ${DELETE_RESOURCES_BEFORE_DATE}"
+echo "--- Installing awscli"
+with_aws_cli
```

Comment:
What is the deadline? So we can add support for other resources in `cloud-reaper`?

Reply:
There is no hard deadline for this. We were planning to merge this PR to start checking (with `plan` command) if there is any stale resource with the current resources supported by `cloud-reaper`.

We were thinking to add in following PRs, at least listing and filtering by tags for the other resources using `aws` CLI directly starting with `redshift` clusters. But, we would prefer to do so also using `cloud-reaper` to keep all the cleanup process using the same tool.

Should we wait for the `cloud-reaper` support instead for the other resources ? Or, should we add the `redshift` listing/filtering with `aws` CLI for now and some notification ? Those `redshift` clusters were the resources that were left in some builds @jsoriano @kpollich

Reply:
> I should be able to add support for those resources.
Thanks @amannocci !

I think probably we could wait for some time, WDYT @jsoriano @kpollich ?

I've just created a PR to list if there is any AWS Redshift cluster stale due to CI builds. That would allow us to receive email notifications if there is any left cluster until that support is added into cloud-reaper:
https://github.com/elastic/elastic-package/pull/1710

Once this support is added into `cloud-reaper`, this logic could be moved to the `cloud-reaper` for sure, so the cleanup is carried out with the same tool for all resources.

In any case, this PR could be merged to start checking there is no resource like instances or queues.